### PR TITLE
Forms refactor

### PR DESCRIPTION
**Pull request by vic-appirio**
ALL previous issues fixed; please review

@tladendo @thabofletcher @nlitwin @parthshah Please review and merge, I have to complete form sizing and tooltips position for settings page;

### CODE REVIEW

File: app/directives/track-toggle/track-toggle.directive.jade [hunk 1]
```diff
@@ -1,7 +1,7 @@
 .track(ng-repeat="track in ['DESIGN', 'DEVELOP', 'DATA_SCIENCE']")
-  hr(class="{{!$index  && 'first'}}")
+  //- hr(class="{{!$index  && 'first'}}")
```
Comment (parthshah) [inline app/directives/track-toggle/track-toggle.directive.jade]: if you dont need this plz delete this row

File: app/settings/edit-profile/edit-profile.jade [hunk 2]
```diff
@@ -21,8 +21,8 @@
                 span(ng-show="!vm.userData.photoURL || !vm.userData.photoURL.length") Add Image
               input(type='file', name='image', on-file-change='on-file-change', id="change-image-input", style="display: none;")
 
-              .file-delete(ng-show="vm.userData.photoURL && vm.userData.photoURL.length")
-                button.tc-btn.tc-btn-secondary.tc-btn-s(ng-click="vm.deleteImage()", type="button") Delete
+              .file-delete
```
Comment (parthshah) [inline app/settings/edit-profile/edit-profile.jade]: why was this removed?

File: assets/css/directives/input-sticky-placeholder.scss [hunk 3]
```diff
@@ -1,17 +1,17 @@
 @import 'topcoder-includes';
 @import '../partials/combined';
 
-.input-sticky-placeholder-container {
-  position: relative;
+// .input-sticky-placeholder-container {
```
Comment (parthshah) [inline assets/css/directives/input-sticky-placeholder.scss]: can we delete this file?

File: assets/css/partials/_tc-mixins.scss [hunk 4]
```diff
@@ -191,18 +194,18 @@
 }
 
 @mixin form-field {
-  padding-left: 10px;
-  margin-bottom: 5px;
-  background-color: #FCFCFC;
-  border: 1px solid #F0F0F0;
-  border-radius: 2px;
-  height: 40px;
+  // padding-left: 10px;
```
Comment (parthshah) [inline assets/css/partials/_tc-mixins.scss]: please delete these commented out lines, unless we need them for reference

File: gulpfile.js [hunk 5]
```diff
@@ -46,13 +46,17 @@ gulp.task('jade', ['clean-html'], function() {
 gulp.task('styles', ['clean-styles'], function() {
   log('Compiling Sass --> CSS');
   var assetPrefix = envConfig.CONSTANTS.ASSET_PREFIX.length ? envConfig.CONSTANTS.ASSET_PREFIX : '/';
+  var sourceMaps       = $.sourcemaps.init();
+  var soureceMapsWrite = $.sourcemaps.write();
 
   return gulp
     .src(config.sass, {base: './'})
     .pipe($.plumber())
+    .pipe(sourceMaps)
```
Comment (parthshah) [inline gulpfile.js]: please include sourceMaps only if the current env is dev or QA.

**Comment by parthshah** [comment]
@vic-appirio thanks this is great. I have a few comments - two main things though -
* Why was ng-show check removed? : ```
file-delete(ng-show="vm.userData.photoURL && vm.userData.photoURL.length")
-                button.tc-btn.tc-btn-secondary.tc-btn-s(ng-click="vm.deleteImage()", type="button") Delete
+              .file-delete
```
* I prefer we include sourceMap in QA & Dev env. We are doing this check for JS files as well.

File: app/settings/edit-profile/edit-profile.jade (hunk 2 above)
Comment (tladendo) [inline app/settings/edit-profile/edit-profile.jade]: +1, this should only be there when there is a profile image present

File: assets/css/account/account.scss [hunk 6]
```diff
@@ -199,7 +181,7 @@
       color: #A3A3AE;
       font-size: 13px;
       line-height: 22px;
-      @include sofia-pro-regular;
+      @include font-with-weight;
```
Comment (tladendo) [inline assets/css/account/account.scss]: this seems way more ambiguous to me but i guess it's fine for now

**Comment by tladendo** [comment]
@vic-appirio -- few things
- password tooltip is weird, gets in the way on settings page
- the "mandatory" field labels look super bunched together with the other text -- maybe the forms should be wider?
- + everything parth said

File: assets/css/partials/_tc-mixins.scss (hunk 4 above)
Comment (vic-appirio) [inline assets/css/partials/_tc-mixins.scss]: I don't want to go full blown remove of mixins yet, this is useful for refactoring yet.
